### PR TITLE
👷🏻‍♂️ Multihop

### PR DESCRIPTION
**Pull request by moss-eth**

### CODE REVIEW

**Comment by barrasso** [comment]
Closes #49

File: src/Zap.sol [hunk 1]
```diff
@@ -361,6 +361,7 @@ contract Zap is Reentrancy, Errors {
         uint128 _collateralId,
         uint256 _collateralAmount,
         address _collateral,
+        bytes memory _path,
```
Comment (ghost) [inline src/Zap.sol:362]: please update natspec with param too 🙏

File: src/Zap.sol [hunk 2]
```diff
@@ -453,61 +455,80 @@ contract Zap is Reentrancy, Errors {
...
+            unwound =
+                _unwrap(_collateralId, _collateralAmount, _unwrapTolerance);
+
+            // establish total debt now owed to Aave;
+            // i.e., # of USDC
+            _flashloan += _premium;
+        }
+
+        (,,, address _collateral, bytes memory _path,,, uint256 _swapTolerance,)
+        = abi.decode(
```
Comment (ghost) [inline src/Zap.sol:508]: why are we decoding this again? was there a stack too deep issue?

File: src/Zap.sol [hunk 3]
```diff
@@ -453,61 +455,80 @@ contract Zap is Reentrancy, Errors {
...
                 uint256,
                 uint256,
                 address
             )
         );
 
-        // zap USDC from flashloan into USDx;
-        // ALL USDC flashloaned from Aave is zapped into USDx
-        uint256 usdxAmount = _zapIn(_flashloan, _zapTolerance);
-
```
Comment (ghost) [inline src/Zap.sol:478]: NIT but will you add space back? lol

File: test/Swap.for.t.sol [hunk 4]
```diff
@@ -17,7 +17,7 @@ import {
 
 contract SwapForTest is Bootstrap {
 
-    function test_swap_for_base() public base {
+    function test_swap_for_single_base() public base {
```
Comment (ghost) [inline test/Swap.for.t.sol:20]: nice nice. good test 🫡

**Review by ghost — APPROVED** [review]
LGTM! Left a few comments, but nice work!

File: src/Zap.sol (hunk 2 above)
Comment (moss-eth) [reply src/Zap.sol:508]: yea exactly

File: src/Zap.sol (hunk 3 above)
Comment (moss-eth) [reply src/Zap.sol:478]: resolved in [bd7f513](https://github.com/JaredBorders/zap/pull/50/commits/bd7f513596aa172a4894e233e83978690e373e55)

File: src/Zap.sol (hunk 1 above)
Comment (moss-eth) [reply src/Zap.sol:362]: resolved in [bd7f513](https://github.com/JaredBorders/zap/pull/50/commits/bd7f513596aa172a4894e233e83978690e373e55)

**Pull request merged**